third_party/xla_client/xrt_computation_client.cc

```diff
@@ -1410,13 +1408,15 @@ std::vector<std::string> XrtComputationClient::GetAllDevices() const {
 }
 
 void XrtComputationClient::SetReplicationDevices(
-    std::vector<std::string> devices) {
-  g_replication_devices = std::move(devices);
+    std::shared_ptr<std::vector<std::string>> devices) {
+  std::lock_guard<std::mutex> lock(lock_);
+  replication_devices_ = std::move(devices);
 }
 
-const std::vector<std::string>& XrtComputationClient::GetReplicationDevices()
-    const {
-  return g_replication_devices;
+std::shared_ptr<std::vector<std::string>>
+XrtComputationClient::GetReplicationDevices() {
+  std::lock_guard<std::mutex> lock(lock_);
+  return replication_devices_;
```

Comment:
Does this means all thread share the same `replication_devices_`?

Reply:
Yes, that is always the case.
The replication devices is a global concept, and I don't even remember why it was TLS in the first place.